packages/@lwc/engine/src/framework/base-lightning-element.ts

```diff
@@ -247,44 +237,48 @@ function BaseLightningElementConstructor(this: LightningElement) {
             `Component creation requires a DOM element to be associated to ${vmBeingConstructed}.`
         );
     }
-    const vm = vmBeingConstructed as VM;
+
+    const vm = vmBeingConstructed;
     const {
         elm,
         mode,
         def: { ctor },
     } = vm;
+
     const component = this;
-    vm.component = component;
-    vm.tro = getTemplateReactiveObserver(vm);
-    vm.oar = create(null);
-    // interaction hooks
-    // We are intentionally hiding this argument from the formal API of LWCElement because
-    // we don't want folks to know about it just yet.
+    const cmpRoot = elm.attachShadow({
+        mode,
+        delegatesFocus: !!ctor.delegatesFocus,
+        '$$lwc-synthetic-mode$$': true,
+    } as any);
+
+    vm.component = this;
+    vm.cmpRoot = cmpRoot;
+
+    // Locker hooks assignment. When the LWC engine is evaluated along Locker, Locker intercepts all
+    // the new component creation and pass hooks to instrument all the component interactions We are
+    // intentionally hiding this argument from the formal API of LWCElement because we don't want
+    // folks to know about it just yet.
```

Comment:
```suggestion
    // Locker hooks assignment. When the LWC engine is run with a Locker layer on top, Locker intercepts all
    // new component creation and passes hooks to instrument the component interactions with the engine.
    // We are intentionally hiding this argument from the formal API of LightningElement because we don't 
    // want folks to know about it just yet.
```